packages/block-editor/src/components/block-list-appender/index.js

```diff
@@ -8,56 +8,61 @@ import { last } from 'lodash';
  */
 import { withSelect } from '@wordpress/data';
 import { getDefaultBlockName } from '@wordpress/blocks';
-import { __ } from '@wordpress/i18n';
-import { IconButton } from '@wordpress/components';
 
 /**
  * Internal dependencies
  */
 import IgnoreNestedEvents from '../ignore-nested-events';
 import DefaultBlockAppender from '../default-block-appender';
-import Inserter from '../inserter';
+import ButtonBlockAppender from '../button-block-appender';
 
 function BlockListAppender( {
 	blockClientIds,
 	rootClientId,
 	canInsertDefaultBlock,
 	isLocked,
+	renderAppender,
 } ) {
 	if ( isLocked ) {
 		return null;
 	}
 
-	if ( canInsertDefaultBlock ) {
+	// If auto-insert Blocks is enabled, default to the standard behaviour
+	// of auto-inserting a Block but only if no renderAppender is provided.
+	if ( ! renderAppender && canInsertDefaultBlock ) {
```

Comment:
Unless I'm mistaken, these return conditions could be simplified:

```jsx
if ( isLocked ) {
	return null;
}

if ( renderAppender ) {
	return renderAppender();
}

if ( canInsertDefaultBlock ) {
	return <DefaultBlockAppender />;
}

return <Inserter />;
```

I was quite confused by the lack of a `return` statement outside of an `if` condition. It seems `Inserter` should be the "default" return assuming (a) not locked, (b) no `renderAppender` prop, and (c) can't insert default block.

There's also another option which could help avoid the duplication (maintenance burden of keeping in sync) of `<div className="block-list-appender">`:

```jsx
let appender;
if ( renderAppender ) {
	appender = renderAppender();
} else if ( canInsertDefaultBlock ) {
	appender = <DefaultBlockAppender />;
} else {
	appender = <Inserter />;
}

return (
	<div className="block-list-appender">
		{ appender }
	</div>
);
```

Reply:
Good suggestion, I went with the first option, I don't think duplicating the div is too bad, it results in a bit more readability.

Reply:
> Good suggestion, I went with the first option, I don't think duplicating the div is too bad, it results in a bit more readability.

My thinking is a bit more in the realm of maintainability, where duplication runs the risk that a future maintainer updates one `div`, when all three of them ought to be updated. (I can't recall specifics, but I definitely recall a similar thing happening in the past)